ext/semian/sysv_semaphores.c

```diff
@@ -130,22 +139,29 @@ get_semaphore(int key)
   return semget(key, SI_NUM_SEMAPHORES, 0);
 }
 
+struct acquire_semaphore_args {
+  semian_resource_t *res;
+  int *wait_time_ms;
+};
+
 void *
-acquire_semaphore_without_gvl(void *p)
+acquire_semaphore_without_gvl(semian_resource_t *res, int *wait_time_ms)
 {
-  WITHOUT_GVL(acquire_semaphore, p, RUBY_UBF_IO, NULL);
+  struct acquire_semaphore_args args = { .res = res, .wait_time_ms = wait_time_ms };
```

Comment:
might be cleaner to just include this as a member of semian_resource_t - and we actually own that type, so we don't have to do any sort of struct-wrapping.